src/pages/dex/[poolpairId]/index.page.tsx

```diff
@@ -19,48 +21,62 @@ interface PoolPairPageProps {
+  ]
+
+  useEffect(() => setTypeCurrentSelection('All'), [props.swaps])
+
   return (
     <>
       <Head
         title='DEX'
         description='Supply liquidity to BTC, ETH, USDT, USDC and many other pool pairs to power the Decentralized Exchange. Earn fees and block rewards in return for providing liquidity to the pool, you can withdraw your liquidity at any time.'
       />
       <Container className='pt-12 pb-20'>
-        <h1 className='text-2xl font-medium mb-6'>
-          <PoolPairSymbolLocal
-            tokenA={props.poolpair.tokenA} tokenB={props.poolpair.tokenB} symbolSizeClassName='h-8 w-8'
-            symbolMarginClassName='ml-5' textClassName='ml-16 font-medium'
-          />
-        </h1>
-
-        <div className='flex flex-wrap -mx-6'>
-          <div className='w-full lg:w-1/2 px-6'>
+        <PoolPairDetailsBar poolpair={props.poolpair} />
+        <div className='flex flex-wrap space-y-12 lg:space-y-0 lg:flex-nowrap mt-8'>
+          <div className='lg:mr-4 w-full lg:w-1/4 min-w-[320px]'>
+            <h3 className='text-lg font-semibold'>
+              Pool Details
+            </h3>
             <PoolPairDetails poolpair={props.poolpair} />
           </div>
-
-          {/* <div className='w-full mt-8 lg:mt-0 lg:w-3/5 xl:w-3/4 px-6'> */}
-          {/*  <PoolPairGraph poolpair={props.poolpair} /> */}
-          {/* </div> */}
-        </div>
-
-        <div className='border-b my-6 border-gray-200' />
-
-        <h3 className='text-xl font-semibold'>
-          Swap History
-        </h3>
-
-        <div className='my-6 hidden md:block'>
-          <SwapTable swaps={props.swaps.items} />
-        </div>
-
-        <div className='my-6 md:hidden'>
-          <SwapCards swaps={props.swaps.items} />
-        </div>
-
-        <div className='flex justify-end mt-8'>
-          <CursorPagination pages={props.swaps.pages} path={`/dex/${props.poolpair.id}`} />
+          <div className='w-full lg:w-3/4 overflow-hidden'>
+            <div className='flex flex-wrap items-center justify-between'>
+              <h3 className='text-lg font-semibold'>
+                Swap History
+              </h3>
+              <div className='flex w-full lg:max-w-max flex-wrap -mx-3 mt-6 md:mt-0 ' data-testid='FeedFilter.Types'>
+                {types.map(type => (
+                  <div
+                    className='flex items-center mx-3  cursor-pointer'
+                    onClick={() => setTypeCurrentSelection(type)}
+                    key={type}
+                    data-testid='SwapFilter.Types'
+                  >
+                    <span className={classNames('rounded border-2 mr-2', typeSelection === type ? 'text-white bg-primary-500 border-primary-500' : 'border-gray-300 ')}>
+                      <IoMdCheckmark className='text-white p-0 m-0 w-4 h-4' />
+                    </span>
+                    <span>{type}</span>
+                  </div>
+                ))}
+              </div>
```

Comment:
```suggestion
              <div className='flex w-full lg:max-w-max flex-wrap -mx-3 mt-6 md:mt-0 ' data-testid='FeedFilter.Types'>
                {types.map(type => (
                  <div
                    className='flex items-center mx-3  cursor-pointer'
                    onClick={() => setTypeCurrentSelection(type)}
                    key={type}
                    data-testid='SwapFilter.Types'
                  >
                    <span className={classNames('rounded border-2 mr-2', typeSelection === type ? 'text-white bg-primary-500 border-primary-500' : 'border-gray-300 ')}>
                      <IoMdCheckmark className='text-white p-0 m-0 w-4 h-4' />
                    </span>
                    <span>{type}</span>
                  </div>
                ))}
              </div>
```
Let's do without the filters first. As it will get really complicated with the composite swaps.

Reply:
Yeah i found out that too on the dFB-dUsd pair